### PR TITLE
Datasource for Equinix Fabric Connection

### PR DESCRIPTION
**Pull request by t0mk**
This PR adds a datasource for connections.

It's from my fork because there are no acceptance tests.

https://metal.equinix.com/developers/docs/networking/fabric/

Part of #11

### CODE REVIEW

File: docs/data-sources/connection.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+page_title: "Equinix Metal: connection"
+subcategory: ""
+description: |-
+  Retrieve Equinix Fabric Connection
+---
+
+# metal\_connection
+
+Use this data source to retrieve a connection resource from [Equnix Fabric - software-defined interconnections](https://metal.equinix.com/developers/docs/networking/fabric/)
```
Comment (displague) [inline docs/data-sources/connection.md:10]: Equinix

File: metal/datasource_metal_connection.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/packethost/packngo"
+)
+
+func connectionPortSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
```
Comment (displague) [inline metal/datasource_metal_connection.go:11]: Should we get in the habit of including the `Description` field, for #13?

Comment (displague) [inline metal/datasource_metal_connection.go:11]: Should we get in the habit of including the `Description` field, for #13?

File: metal/datasource_metal_connection.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	return ret
+}
+
+func dataSourceMetalConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*packngo.Client)
+	connId := d.Get("connection_id").(string)
+
+	conn, _, err := client.Connections.Get(
+		connId,
+		&packngo.GetOptions{Includes: []string{"organization,facility"}})
```
Comment (displague) [inline metal/datasource_metal_connection.go:122]:
```suggestion
		&packngo.GetOptions{Includes: []string{"organization","facility"}})
```

**Comment by displague** [comment]
Let me know when this is ready for review. I'll have to make sure our testing account can work with Fabric connection resources in some way.

**Comment by t0mk** [comment]
@displague this is ready for review. 

There are no acceptance tests, because we can't create and destroy a connection resource - it needs to be requested and request must be approved. We could theoretically create a connection resource in the testing account just for acceptance test, but I'm not sure if this is justified.

File: metal/datasource_metal_connection.go (hunk 2 above)
Comment (t0mk) [reply metal/datasource_metal_connection.go:11]: yes, probably good idea. Just it means writing it twice until we are able to finish #13

File: docs/data-sources/connection.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+* `type` - Connection type, dedicated or shared
+* `redundancy` - Connection redundancy, reduntant or primary
+* `speed` - Connection speed in bits per second
+* `ports` - List of connection ports - primary (`ports[0]`) and secondary (`ports[1]`)
+  * `name` - Port name
+  * `role` - Port role - primary or secondary
+  * `speed` - Port speed in bits per second
+  * `status` - Port status 
+  * `link_status` - Port link status
+  * `virtual_circuit_ids` - List of IDs of virtual cicruits attached to this port
```
Comment (displague) [inline docs/data-sources/connection.md:41]: typo - circuits

**Pull request merged**